types/webextension-polyfill/namespaces/runtime.d.ts

```diff
@@ -343,10 +471,14 @@ export namespace Runtime {
          * @param extensionId Optional. The ID of the extension/app to send the message to. If omitted,
          * the message will be sent to your own extension/app. Required if sending messages from a web page for
          * $(topic:manifest/externally_connectable)[web messaging].
-         * @param message
          * @param options Optional.
          */
-        sendMessage(extensionId: string | undefined, message: any, options?: SendMessageOptionsType): Promise<any>;
+        // eslint-disable-next-line @definitelytyped/no-unnecessary-generics
+        sendMessage<TMessage = unknown, TResponse = unknown>(
```

Comment:
This seems like another change from `any` to `unknown` in an output position, since inference of type arguments will default `TResponse` to `unknown`. In addition to that, `TResponse` is only used in the return type, so inference will not work as it doesn't have a place to infer from. And `TMessage` seems useless a similar same reason, it only shows up in the parameter type.

Reply:
Inference is not an option here. This will send out a message and the listener can return something that might not be related to the message itself.

There are multiple solutions to this problem:

1. don't use generic types at all and use (promise of) unknown as return type, which essentially means you can pass anything. Users want to be able to add a type-check here (it was an actual feature request).
2. only use generic type for the message parameter. You'd still have to manually cast the return value.
3. The proposed solution

Here are the different solutions in code:
```ts
// 1: 
const x = (await sendMessage(extId, icanPassWhatIWant)) as MyReturnType;
// 2:
const x = (await sendMessage<MyMsgType>(extId, msg)) as MyReturnType;
// 3:
const x = await sendMessage<MyMsgType, MyReturnType>(extId, msg);
```

I find 3 to be the best of them.